berkeley-mobile/StudyPact/ProfileViewController.swift

```diff
@@ -32,16 +38,53 @@ class ProfileViewController: UIViewController, UITextFieldDelegate, UINavigation
         self.view.backgroundColor = Color.modalBackground
         
         setupHeader()
-        setupProfile()
         
-        let tap = UITapGestureRecognizer(target: self, action: #selector(UIInputViewController.dismissKeyboard))
-        view.addGestureRecognizer(tap)
+        GIDSignIn.sharedInstance()?.delegate = self
         
-        imagePicker = UIImagePickerController()
-        imagePicker.delegate = self
-        imagePicker.sourceType = .photoLibrary
+        if GIDSignIn.sharedInstance().hasPreviousSignIn() {
+            // Signed in
+
+            GIDSignIn.sharedInstance()?.restorePreviousSignIn()
+            loggedInView()
+        } else {
+            GIDSignIn.sharedInstance()?.presentingViewController = self
+            
+            loggedOutView()
+        }
+    }
+    
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error!) {
+        if let error = error {
+            print(error)
+            return
+        }
+        guard let authentication = user.authentication else {
+            loggedOutView()
+            return
+        }
+        // TODO: - Store the useful hash?
+        let credential = GoogleAuthProvider.credential(withIDToken: authentication.idToken,
+                                                       accessToken: authentication.accessToken)
         
-        setupProfileImage()
+        loggedInView()
+        
+        let userId = user.userID  // Client side only
+        let idToken = user.authentication.idToken // Safe to send to the server
+        let fullName = user.profile.name
+        let email = user.profile.email
+        
+        initialsLabel.text = Array(arrayLiteral: fullName)[0]
+        fullNameField.text = fullName
+        emailTextField.textField.text = email
+        
+        if user.profile.hasImage {
+            guard let imageUrl = user.profile.imageURL(withDimension: 100) else { return }
```

Comment:
bump the 100 up a bit, my profile pic from gmail (default single letter) was very pixelated. 200 looked fine